File: sprint3/go/lesson2code4.go
func asBinaryString(n int) string {
    if n < 0 {
        return "-" + asBinaryString(-n)
    }
    if n == 0 || n == 1 {
        return strconv.Itoa(n)
    }
    lastDigit := n % 2
    return asBinaryString(n / 2) + strconv.Itoa(lastDigit)
}

File: sprint3/JS/lesson2code4.js
function as_binary_string(n) {
    if (n < 0) {
        return "-" + as_binary_string(-n);
    }
    if (n == 0 || n == 1) {
        return n.toString();
    }
    let last_digit = n % 2;
    return as_binary_string(Math.floor(n / 2)) + last_digit.toString();
}
